extensions/cornerstone/src/services/ViewportService/CornerstoneViewportService.ts

```diff
@@ -163,6 +170,64 @@ class CornerstoneViewportService implements IViewportService {
 
     this.viewportsInfo.get(viewportIndex).destroy();
     this.viewportsInfo.delete(viewportIndex);
+    this.viewportsById.delete(viewportId);
+  }
+
+  public getPresentation(viewportIdx: number): Presentation {
+    const viewportInfo = this.viewportsInfo.get(viewportIdx);
+    if (!viewportInfo) return;
+    const {
+      presentationId: id,
+      viewportType,
+    } = viewportInfo.getViewportOptions();
+    if (!id) return;
+
+    const csViewport = this.getCornerstoneViewportByIndex(viewportIdx);
+    if (!csViewport) return;
+
+    if (viewportType === 'stack' || !viewportType) {
+      const pan = csViewport.getPan();
+      const zoom = csViewport.getZoom();
+      const properties = csViewport.getProperties();
+      const initialImageIndex = csViewport.getCurrentImageIdIndex();
+      return {
+        id,
+        viewportType: viewportType === 'volume3d' ? 'volume' : 'stack',
```

Comment:
if it is `viewportType===stack`, why it can be `volume3D` or `volume`?

Reply:
Because CS3D redefined the viewportType definitions, and it no longer sets the right type value otherwise.  Really we should be defining two keywords, or else completely synchronizing them, but I don't really want to do that in this PR.